src/Doxyfile.diff

```diff
@@ -0,0 +1,84 @@
+# The format is:
+# TAG = value [value, ...]
+# For lists, items can also be appended using:
+# TAG += value [value, ...]
+# Values that contain spaces should be placed between quotes (\" \").
+
+#---------------------------------------------------------------------------
+# Project related configuration options
+#---------------------------------------------------------------------------
+
+# The PROJECT_NAME tag is a single word (or a sequence of words surrounded by
+# double-quotes, unless you are using Doxywizard) that should identify the
+# project for which the documentation is generated. This name is used in the
+# title of most generated pages and in a few other places.
+# The default value is: My Project.
+
+PROJECT_NAME           = "metamath executable"
+
+# Set to Metamath version
+
+# The PROJECT_NUMBER tag can be used to enter a project or revision number. This
+# could be handy for archiving the generated documentation or if some version
+# control system is used.
+
+PROJECT_NUMBER         = "Metamath-version"
+
+# Using the PROJECT_BRIEF tag one can provide an optional one line description
+# for a project that appears at the top of each page and should give viewer a
+# quick idea about the purpose of the project. Keep the description short.
+
+PROJECT_BRIEF          = "executable used to maintain *.mm files" 
+
+# With the PROJECT_LOGO tag one can specify a logo or an icon that is included
+# in the documentation. The maximum height of the logo should not exceed 55
+# pixels and the maximum width should not exceed 200 pixels. Doxygen will copy
+# the logo to the output directory.
+
+PROJECT_LOGO           = "Metamath.png"
+
+# The OUTPUT_DIRECTORY tag is used to specify the (relative or absolute) path
+# into which the generated documentation will be written. If a relative path is
+# entered, it will be relative to the location where doxygen was started. If
+# left blank the current directory will be used.
+
+OUTPUT_DIRECTORY       = "../doxy"
+
+#---------------------------------------------------------------------------
+# Configuration options related to the dot tool
+#---------------------------------------------------------------------------
+
+# If you set the HAVE_DOT tag to YES then doxygen will assume the dot tool is
+# available from the path. This tool is part of Graphviz (see:
+# http://www.graphviz.org/), a graph visualization toolkit from AT&T and Lucent
+# Bell Labs. The other options in this section have no effect if this option is
+# set to NO
+# The default value is: YES.
+
+HAVE_DOT               = NO
+
+#---------------------------------------------------------------------------
+# Configuration options related to the LaTeX output
+#---------------------------------------------------------------------------
+
+# If the GENERATE_LATEX tag is set to YES, doxygen will generate LaTeX output.
+# The default value is: YES.
+
+GENERATE_LATEX         = NO
+
```

Comment:
```suggestion
# diff to Doxyfile 1.8.17 with only default settings.

PROJECT_NAME = "Metamath Executable"
# TODO: Set to Metamath version
PROJECT_NUMBER = "Metamath-version"
PROJECT_BRIEF = "executable used to maintain *.mm files" 
PROJECT_LOGO = "Metamath.png"
OUTPUT_DIRECTORY = "../doxy"
# disable Graphviz .dot output
HAVE_DOT = NO
# disable LaTeX output
GENERATE_LATEX = NO
```

Reply:
The people likely editing this file are developers (like you) who want to tweak source documentation.  Say you like graphs in your output and have dot installed.  Why should this information `doxygen will assume the dot tool is
available from the path` be hidden from you?

Some of the changes I made in the past few days target the inexperienced developer, not the expert.  Annotating configure.ac for example.  Experts will likely be annoyed by the comments, that most often tell them nothing new and just clobber up the editor contents.  Yes.  I know.  But it is a matter of a few seconds to delete the comments and get a concise view.  By contrast, it is a matter of hours to wade through documentation hundreds of pages thick to find out what is going on for a newbie.  Statistics should reflect the probability of what kind your average reader is.  What do you think is the trade off then?

That said, I am not really picky about this point.  I can live with both results.

Reply:
I was assuming that the dot setting was just a "no, this is inapplicable for us", and we don't need to duplicate doxygen documentation for someone who really wants to try an unsupported configuration. If you actually want to consider setting that to true, then sure, it would make sense to explain what else has to be done in order to turn that feature on.

I have nothing against documentation for the inexperienced developer, but it should be *our* documentation, recontextualizing the information in the context of this project. It's pretty obvious what `PROJECT_LOGO` does from the name and the value it is set to, and I think that it is actually much clearer to remove the wall of text spreading the three lines of interesting configuration across two pages.

Reply:
Apart from project name and number nothing is fixed so far in the doxygen configuration.  If others think my choices should be modified, yep, go ahead.  Note that the current state of affairs is **setting up** a framework useful to developers only (not end users).  While this set up shows points of commonality with the general build process and should be finally an integral part of it, I am not primarily bothered with the general build (apart from documenting it).  I want to get the doxygen documentation of the C sources off the ground.  To this end build.sh/Doxyfile.diff are convenience installations providing fast results in the current environment.  For testing, suggesting improvements and so on.  If they vanish after a few weeks and are replaced with something better - that is fine with me.

Reply:
Okay. Frankly I don't know why you are spending so much time on doc setup and meta-documentation instead of actually writing documentation at this point. You don't need doxygen to get use out of doxygen style documentation - lean has something that resembles doxygen documentation but I have never seen it built or attempted to build it myself. It's still plenty readable as documentation.

I still think you should remove all the garbage from the doxygen.diff file. If you want to have comments in it, okay, but the comment ratio should not go beyond 2-1.

Reply:
>...why you are spending so much time on doc setup and meta-documentation instead of actually writing documentation at this point.

First, I guess, reviewing your code cost me about the same time.

Then, it is my sincere hope that I will not be the only one documenting the sources.  In private I had a system up for a few weeks now.  The problem is to get support of others, too.  You need to agree on a common style, find a way to work together, see where the limits are, and so on.  In the beginning I suggested AsciiDoc, now we use Doxygen.  This is a clear indication that my ideas need some review, and are not accepted out of the box.

The meta-documentation is just an example on how things can be done.  If people don't like the style, I hopefully learn soon.

As for the Doxyfile, I already accepted all of your suggestions.  I prefer getting on over disputing the comment style for more than a couple of hours.

Reply:
I thought resolving the conversation automatically accepts your suggestion.  It does not.  I manually updated the file now.